From 707398814bf924b6c430a49897171d1322011bff Mon Sep 17 00:00:00 2001
From: Kevin Wolf <kwolf@redhat.com>
Date: Thu, 26 Sep 2013 19:55:33 +0800
Subject: [PATCH 18/37] block: use DIV_ROUND_UP in bdrv_co_do_readv

Message-id: <1392117622-28812-19-git-send-email-kwolf@redhat.com>
Patchwork-id: 57183
O-Subject: [RHEL-7.0 qemu-kvm PATCH v2 18/37] block: use DIV_ROUND_UP in bdrv_co_do_readv
Bugzilla: 748906
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>
RH-Acked-by: Max Reitz <mreitz@redhat.com>

From: Fam Zheng <famz@redhat.com>

Signed-off-by: Fam Zheng <famz@redhat.com>
Reviewed-by: Eric Blake <eblake@redhat.com>
Signed-off-by: Kevin Wolf <kwolf@redhat.com>
(cherry picked from commit d055a1fec37ec84fd3e87d48a0e766a9ff8369c4)

Signed-off-by: Kevin Wolf <kwolf@redhat.com>
---
 block.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
---
 block.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/block.c b/block.c
index 645bcd0..feeab4e 100644
--- a/block.c
+++ b/block.c
@@ -2795,7 +2795,7 @@ static int coroutine_fn bdrv_aligned_preadv(BlockDriverState *bs,
             goto out;
         }
 
-        total_sectors = (len + BDRV_SECTOR_SIZE - 1) >> BDRV_SECTOR_BITS;
+        total_sectors = DIV_ROUND_UP(len, BDRV_SECTOR_SIZE);
         max_nb_sectors = MAX(0, total_sectors - sector_num);
         if (max_nb_sectors > 0) {
             ret = drv->bdrv_co_readv(bs, sector_num,
-- 
1.7.1

